pallets/handles/src/handles-utils/src/validator.rs

```diff
@@ -10,7 +10,21 @@ use constants::ALLOWED_UNICODE_CHARACTER_RANGES;
 
 /// Reserved words that cannot be used as the handle.
 const RESERVED_WORDS: [&str; 8] =
-	["admin", "everyone", "all", "administrator", "mod", "moderator", "here", "channel"];
+	["adm1n", "every0ne", "a11", "adm1n1strat0r", "m0d", "m0derat0r", "here", "channe1"];
+
+// We MUST have the RESERVED_WORDS constant as canonical.
+// Cannot easily be canonicalized at compile time currently
+#[test]
+fn ensure_reserved_words_canonical() {
+	for &word in &RESERVED_WORDS {
+		let canonical = crate::convert_to_canonical(word);
+		assert_eq!(
+			word, canonical,
+			"The reserved word '{}' MUST match canonical form: '{}'",
+			word, canonical
+		);
+	}
+}
```

Comment:
not sure if this would be a better test or not; otherwise technically the test passes if even if `convert_to_canonical` is a no-op
```suggestion

// We MUST have the RESERVED_WORDS constant as canonical.
// Cannot easily be canonicalized at compile time currently
#[test]
fn ensure_reserved_words_canonical() {
    const TEST_WORDS: [&str; 8] =
	["admin", "everyone", "all", "administrator", "mod", "moderator", "here", "channel"];

	for (i, &word) in TEST_WORDS.iter().enumerate() {
		let canonical = crate::convert_to_canonical(word);
		assert_eq!(
			RESERVED_WORDS[i], canonical,
			"The reserved word '{}' MUST match canonical form: '{}'",
			word, canonical
		);
	}
}
```

Reply:
True. This is just a check to make sure that `convert_to_canonical` does NOT change the output of `RESERVED_WORDS`. Other test (`test_is_reserved_canonical_handle_happy_path) make sure that reserved words are rejected.